src/main/java/seedu/mark/logic/commands/DeleteCacheCommand.java

```diff
@@ -0,0 +1,69 @@
+package seedu.mark.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collections;
+import java.util.List;
+
+import seedu.mark.commons.core.Messages;
+import seedu.mark.commons.core.index.Index;
+import seedu.mark.logic.commands.exceptions.CommandException;
+import seedu.mark.logic.commands.results.CommandResult;
+import seedu.mark.model.Model;
+import seedu.mark.model.bookmark.Bookmark;
+import seedu.mark.storage.Storage;
+
+/**
+ * Deletes an offline copy of a bookmark.
+ */
+public class DeleteCacheCommand extends Command {
+
+    public static final String COMMAND_WORD = "cache-delete";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes the offline copy of a bookmark.\n"
+            + "Parameters: INDEX";
+
+    public static final String MESSAGE_SUCCESS = "Bookmark's offline copy successfully deleted!";
+    public static final String MESSAGE_FAILURE =
+            "No offline copy is available to delete!!";
+
+    private Index index;
+
+    /**
+     * Instantiates a new DeleteCache Command.
+     *
+     * @param index the index to delete
+     */
+    public DeleteCacheCommand(Index index) {
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model, Storage storage) throws CommandException {
+        requireNonNull(model);
+        List<Bookmark> lastShownList = model.getFilteredBookmarkList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_BOOKMARK_DISPLAYED_INDEX);
+        }
+
+        Bookmark bookmarkToDelete = lastShownList.get(index.getZeroBased());
+
+        if (bookmarkToDelete.getCachedCopies().isEmpty()) {
+            throw new CommandException(MESSAGE_FAILURE);
+        }
+
+        Bookmark updatedBookmark = new Bookmark(bookmarkToDelete.getName(), bookmarkToDelete.getUrl(),
```

Comment:
Would it be easier to use `BookmarkBuilder` here?

Reply:
Nope, because bookmark build can only add cached  copies, not delete them